src/components/dialogHelper/dialogHelper.js

```diff
@@ -144,9 +148,19 @@ import '../../assets/css/scrollstyles.scss';
         animateDialogOpen(dlg);
 
         if (isHistoryEnabled(dlg)) {
-            appRouter.show(`/dialog?dlg=${hash}`, { dialogId: hash });
+            const state = history.location.state || {};
+            const dialogs = state.dialogs || [];
+            dialogs.push(hash);
+
+            history.push(
+                `${history.location.pathname}${history.location.search}`,
```

Comment:
`Relative pathnames are not supported in hash history.push("!/home.html")` warning. Could this lead to that behavior?

Reply:
`appRouter.show` is promised, so the previous `back` is handled before opening a new dialog. This is only for history, the dialog is already there, iirc.